api-stat.go

```diff
@@ -187,6 +198,7 @@ func (c Client) statObject(ctx context.Context, bucketName, objectName string, o
 		// Extract only the relevant header keys describing the object.
 		// following function filters out a list of standard set of keys
 		// which are not part of object metadata.
-		Metadata: extractObjMetadata(resp.Header),
+		Metadata:     metadata,
+		UserMetadata: userMetadata,
```

Comment:
Instead of a separate UserMetadata map, we can just strip the x-amz-meta- prefix in extractObjMetadata.